server/data/data-jmap/src/main/java/org/apache/james/jmap/memory/vacation/MemoryVacationRepository.java

```diff
@@ -44,10 +46,31 @@ public MemoryVacationRepository() {
     }
 
     @Override
-    public CompletableFuture<Void> modifyVacation(AccountId accountId, Vacation vacation) {
+    public CompletableFuture<Void> modifyVacation(AccountId accountId, VacationUpdate vacationUpdate) {
         Preconditions.checkNotNull(accountId);
-        Preconditions.checkNotNull(vacation);
-        vacationMap.put(accountId, vacation);
+        Preconditions.checkNotNull(vacationUpdate);
+        Vacation oldVacation = retrieveVacation(accountId).join();
+        vacationMap.put(accountId, applyUpdate(oldVacation, vacationUpdate));
         return CompletableFuture.completedFuture(null);
     }
+
+    private Vacation applyUpdate(Vacation vacation, VacationUpdate update) {
+        return Vacation.builder()
```

Comment:
VacationPatch (my take about VacationUpdate right naming) should have a `.patch(Vacation)` method